plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotMetadata.java

```diff
@@ -254,9 +256,31 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
     {
         PinotTableHandle handle = (PinotTableHandle) table;
         TupleDomain<ColumnHandle> oldDomain = handle.getConstraint();
-        // Pinot does not support array literals
-        TupleDomain<ColumnHandle> newDomain = oldDomain.intersect(constraint.getSummary())
-                .filter((columnHandle, domain) -> !(((PinotColumnHandle) columnHandle).getDataType() instanceof ArrayType));
+
+        // Extracted from DefaultJdbcMetadata
+        TupleDomain<ColumnHandle> newDomain = oldDomain.intersect(constraint.getSummary());
+        TupleDomain<ColumnHandle> remainingFilter;
+        if (newDomain.isNone()) {
+            remainingFilter = TupleDomain.all();
+        }
+        else {
+            Map<ColumnHandle, Domain> domains = newDomain.getDomains().orElseThrow();
+
+            Map<ColumnHandle, Domain> supported = new HashMap<>();
+            Map<ColumnHandle, Domain> unsupported = new HashMap<>();
+            for (Map.Entry<ColumnHandle, Domain> entry : domains.entrySet()) {
+                // Pinot does not support array literals
+                if (((PinotColumnHandle) entry.getKey()).getDataType() instanceof ArrayType) {
+                    unsupported.put(entry.getKey(), entry.getValue());
+                }
+                else {
+                    supported.put(entry.getKey(), entry.getValue());
```

Comment:
Yes, there are tests. Case sensitivity is not an issue. Recently boolean and timestamp types were implemented in pinot. In a subsequent pr I will update to use those tests since the types required (boolean and date/timestamp) will be supported.